src/lsm/lsm_worker.c

```diff
@@ -93,11 +92,13 @@ __lsm_worker(void *arg)
 
 	cookie = (WT_LSM_WORKER_ARGS *)arg;
 	session = cookie->session;
-	conn = S2C(session);
 
 	entry = NULL;
-	while (F_ISSET(conn, WT_CONN_SERVER_RUN) &&
-	    F_ISSET(cookie, WT_LSM_WORKER_RUN)) {
+	for (;;) {
+		WT_BARRIER();			/* Don't cache the run flag. */
```

Comment:
Can I ask a different question: What problem are you trying to fix with the barrier? Are you worried that the result of `F_ISSET(cookie, WT_LSM_WORKER_RUN)` will be cached across passes in this loop, and thus we won't exit because we don't see the state change? I understand your dislike of the `volatile` keyword, but that seems like exactly the case `volatile` is designed for.

A quick `grep -rnI "while.*F_ISSET" src/` reveals a dozen places where we use this same structure - I'd much rather not have this `WT_BARRIER` here - it's going to make me think every time I encounter the code.